### PR TITLE
AVM:  Consolidate TEAL and AVM versioning

### PR DESCRIPTION
**Pull request by michaeldiamant**
Consolidates TEAL and AVM versioning references as requested in https://github.com/algorand/go-algorand/issues/4125.

Diffs produced by:
* Searching for (case insensitive) "teal version", "tealversion", and "teal v" references.
* Reviewing the spec (`data/transactions/logic/README.md`).

### CODE REVIEW

**Review by jannotti — APPROVED** [review]
Looks good

File: data/transactions/logic/eval.go [hunk 1]
```diff
@@ -171,13 +171,13 @@ func stackValueFromTealValue(tv basics.TealValue) (sv stackValue, err error) {
 	return
 }
 
-// ComputeMinTealVersion calculates the minimum safe TEAL version that may be
+// ComputeMinAvmVersion calculates the minimum safe AVM version that may be
 // used by a transaction in this group. It is important to prevent
 // newly-introduced transaction fields from breaking assumptions made by older
 // versions of TEAL. If one of the transactions in a group will execute a TEAL
```
Comment (iten-alg) [inline data/transactions/logic/eval.go:177]:
```suggestion
// versions of the AVM. If one of the transactions in a group will execute a TEAL
```

File: data/transactions/logic/eval_test.go [hunk 2]
```diff
@@ -185,14 +185,14 @@ func TestEmptyProgram(t *testing.T) {
...
+func TestMinAvmVersionParamEvalCheckSignature(t *testing.T) {
 	partitiontest.PartitionTest(t)
 
 	t.Parallel()
 	params := defaultEvalParams(nil)
 	version2 := uint64(rekeyingEnabledVersion)
-	params.MinTealVersion = &version2
+	params.MinAvmVersion = &version2
 	program := make([]byte, binary.MaxVarintLen64)
 	// set the teal program version to 1
```
Comment (iten-alg) [inline data/transactions/logic/eval_test.go:197]:
```suggestion
	// set the program version to 1
```

Comment (michaeldiamant) [reply data/transactions/logic/eval_test.go]: @iten-alg Thanks for catching multiple stale references + supplying _suggestions_. ☕

**Review by iten-alg — APPROVED** [review]
Looks good, couple lines that are maybe relevant (if we decide these are I can go find similar ones):
->line 678 of opcodes.go says "v2 TEAL"
->line 17 of rekey.sh says "v2 TEAL"
Sorry I couldn't find a way in github to add suggestions to these lines.

**Comment by michaeldiamant** [comment]
> couple lines that are maybe relevant 

@iten-alg Thanks for catching - I updated via https://github.com/algorand/go-algorand/pull/4269/commits/2615784b538b35b34a36c7f67d73193e020d5bee using `git --no-pager grep -P "v\d TEAL"` to catch references.

**Review by algoanne — APPROVED** [review]

**Pull request merged**